R/plot.complexFeatures.R

```diff
@@ -1,152 +1,254 @@
-      theme_bw() +
-      theme(panel.grid.major = element_blank(),
-      panel.grid.minor = element_blank()) +
-      guides(fill=FALSE,linetype=FALSE) +
-      theme(plot.margin = unit(c(1.5,.5,.5,.5),"cm")) +
-      theme(plot.title = element_text(vjust=19,size=10))
-
-    if (log) {
-        p <- p + scale_y_log10('log(intensity)')
+    title <- paste0(complexName,
+                    "\nAnnotated subunits: ",n_annotatedSubunits,
+                    "   Subunits with signal: ",n_signalSubunits,
+                    "\nMax. coeluting subunits: ",n_detectedSubunits,
+                    "   Max. completeness: ",complexCompleteness)
+  } else {
+    features <- subset(features, protein_id == feature_id)
+    proteins <- unique(unlist(strsplit(features$subunits_annotated, split = ";")))
+    traces <- subset(traces, trace_subset_ids = proteins)
+    if (annotationID %in% names(traces$trace_annotation)) {
+      complexName = traces$trace_annotation[,get(annotationID)][1]
+    } else {
+      complexName = traces$trace_annotation$protein_id[1]
     }
-
-#    p + geom_vline(aes(xintercept=left_sec, color=subgroup),
-#                   data=found.features) +
-#        geom_vline(aes(xintercept=right_sec, color=subgroup),
-#                   data=found.features) +
-#        theme(legend.position='none')
-
-    if(peak_area==TRUE){
-      p <- p + geom_rect(data=features,aes(xmin = left_pp, xmax = right_pp, ymin = -Inf, ymax = Inf, fill=subunits_detected),alpha = 0.25)
+    n_annotatedSubunits = features$n_subunits_annotated[1]
+    n_detectedSubunits = features$n_subunits_detected[1]
+    complexCompleteness = round(features$completeness[1],digits=2)
+    title <- paste0(complexName,
+                    "\nAnnotated subunits: ",n_annotatedSubunits,
+                    "\nMax. coeluting subunits: ",n_detectedSubunits,
+                    "   Max. completeness: ",complexCompleteness)
+  }
+  if (onlyBest) {
+    features <- getBestFeatures(features)
+  }
+  traces.long <- toLongFormat(traces$traces)
+  traces.long <- merge(traces.long,traces$fraction_annotation,by.x="fraction",by.y="id")
+  
+  ## Get traces to highlight
+  if(!is.null(highlight)){
+    traces.long$outlier <- gsub("\\(.*?\\)","",traces.long$id) %in% gsub("\\(.*?\\)","",highlight)
+  }
+  
+  if (annotationID %in% names(traces$trace_annotation)) {
+    traces.long <- merge(traces.long,traces$trace_annotation,by="id",all.x=TRUE,all.y=FALSE)
+    if (traces$trace_type == "protein") {
+      traces.long[,id := get(annotationID)]
+      proteins <- traces$trace_annotation[match(proteins,traces$trace_annotation$id)][,get(annotationID)]
     }
-    if(sliding_window_area==TRUE){
-      p <- p + geom_vline(data=features,aes(xintercept=left_sw), colour="grey",linetype="longdash")
-      p <- p + geom_vline(data=features,aes(xintercept=right_sw), colour="grey",linetype="longdash")
+    if ("protein_mw" %in% names(traces$trace_annotation)) {
+      if (!is.null(calibration)) {
```

Comment:
Yes I tried this quickly yesterday, but it didn't work because of a few stupid things. Would be nice to do better in the future, but this would take time now, so I would suggest do put this on the nice to have list... :/